### PR TITLE
Add SELinux support with an optional loadable policy

### PR DESCRIPTION
**Pull request by djjudas21**
I ran into some problems with SELinux denying logrotate permissions to actually rotate the logs on CentOS 6. This optional loadable policy should fix that.

Default behaviour is to _not_ load the policy, so there will be no surprises for users who upgrade.

### CODE REVIEW

**Comment by djjudas21** [comment]
Hey, any progress on this one? Thanks!

**Comment by djjudas21** [comment]
I've resynced my fork so it this PR no longer conflicts with `master`. However the CI builds are now failing, but it looks like it's a problem with the tests rather than the code itself. Something about a Rubocop config?

**Comment by bastelfreak** [comment]
Hi @djjudas21, could you please rebase after https://github.com/voxpupuli/puppet-logrotate/pull/63 got merged? We recently migrated this module to Vox Pupuli.

File: files/site-logrotate.te [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module site-logrotate 1.0;
+
+require {
+	type logrotate_t;
+	type var_lib_t;
+	class file { write rename create unlink setattr };
+}
+
+#============= logrotate_t ==============
+allow logrotate_t var_lib_t:file { write rename create unlink setattr };
```
Comment (vinzent) [inline files/site-logrotate.te]: `var_lib_t` is a generic type. `logrotate_t`should not be able to write to it.

**Comment by vinzent** [comment]
@djjudas21 I suspect you're trying to manage custom folders in `/var/lib` like `/var/lib/something/*`?  If this is the case the correct way would be to create a custom policy for the something folder and grant permissions for logrotate. 

Anyway - if you still think it's a good idea to allow `logrotate_t` to manage `var_lib_t` then I suggest you ask to include this in upstream (but I don't think they wil agree with you):

For Fedora (is also the RHEL/CentOS upstream): https://github.com/fedora-selinux/selinux-policy-contrib/blob/rawhide/logrotate.te (malinglist: https://lists.fedoraproject.org/archives/list/selinux@lists.fedoraproject.org/

For the refpolicy (anything not Fedora based): https://github.com/TresysTechnology/refpolicy-contrib/blob/master/logrotate.te (mailinglist: http://oss.tresys.com/mailman/listinfo/refpolicy)

**Comment by djjudas21** [comment]
We're not managing any log files anywhere other than `/var/log` and we haven't changed any SELinux contexts, either - so I'd be surprised if this isn't affecting other people too.

However, I do agree that permitting `logrotate_t` to manage `var_lib_t` is probably a bad idea, so I'll keep my SELinux policy local and put it in a profile insteed. Cheers :+1:

**Comment by vinzent** [comment]
@djjudas21 `var_lib_t` isn't a type configured in /var/log:

```
$ sudo semanage fcontext --list | grep /var/log | grep var_lib_t

```

i'd recommend then to run `restorecon -rv /var/log` to reset selinux labels and investigate what caused wrong labels inside /var/log

**Pull request closed without merging**